src/Cameras/Inputs/babylon.freeCameraMouseInput.ts

```diff
@@ -76,7 +76,7 @@ module BABYLON {
                         var offsetX = evt.clientX - this.previousPosition.x;
                         var offsetY = evt.clientY - this.previousPosition.y;
 
-                        if (this.camera.getScene().useRightHandedSystem) {
+                        if (this.camera.getScene().useRightHandedSystem === (this.camera.parent && this.camera.parent._getWorldMatrixDeterminant() >= 0)) {

```

Comment:
I wonder if this will  or be executed if camera.parent == null or undefined

Reply:
It doesn't because of javascriptism, but I agree it's not clear.